crypto/async/async_wait.c

```diff
@@ -113,22 +113,25 @@ int ASYNC_WAIT_CTX_get_changed_fds(ASYNC_WAIT_CTX *ctx, OSSL_ASYNC_FD *addfd,
 {
     struct fd_lookup_st *curr;
 
-    *numaddfds = ctx->numadd;
-    *numdelfds = ctx->numdel;
```

Comment:
Thinking about it wouldn't it be better to modify ASYNC_WAIT_CTX_clear_fd() to remove `curr` from the list in the event that `curr->add == 1` and decrement `ctx->numadd`? That way `ctx->numadd` and `ctx->numdel` are correct and we don't need to loop through the whole list just to get the numbers.